editor/plugins/script_text_editor.cpp

```diff
@@ -1473,6 +1473,8 @@ void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data
 	int row, col;
 	te->_get_mouse_pos(p_point, row, col);
 
+	bool dropAtCursor = (col <= te->cursor_get_column() + 3 && col >= te->cursor_get_column() - 3 && row == te->cursor_get_line()) ? true : false;
```

Comment:
We use `snake_case` for identifiers.

Reply:
A comment explaining why we do this ±3 would also be good.

Reply:
I honestly don't think it's a good idea. It feels like it works nicely when you have the knowledge of how it is supposed to work, but if you don't know that it's based on the cursor position it might feel like a bug. For me it feels like one or the other is more intuitive. Drag at mouse position feels fine. Drag at cursor feels fine, too. But if you drag at cursor, only when the mouse is near the cursor it doesn't feel right because if you're gonna position the cursor at a certain spot, you should be able to drag anywhere on the editor and not be forced to be specifically within the range of the cursor. For example, what if you're typing along and the cursor is at the end of the line... then you try dragging between the parenthesis, which is at the end of the line, too... it'll position at the very end of the line and not betweeen the parenthesis and since you don't even know that it's based on the cursor, it feels clunky.

As you said before, if there are visible cues that it snaps to the cursor, it probably would make more sense. Until then, I think this small fix more confusing. I felt like this fix would be a better alternative until implementing that, but it's worse than if we had an option for drag at cursor or at mouse position.

Reply:
Does it feel better if you decrease the tolerance to 1 or 2 characters (from 3)?